jobs/payment-jobs/tasks/electronic_funds_transfer_task.py

```diff
@@ -12,273 +12,150 @@
 # limitations under the License.
 """Task for linking electronic funds transfers."""
-from dataclasses import dataclass
-from datetime import datetime
+from datetime import datetime, timezone
 from typing import List
 
 from flask import current_app
 from pay_api.models import CfsAccount as CfsAccountModel
 from pay_api.models import EFTCredit as EFTCreditModel
-from pay_api.models import EFTShortnameLinks as EFTShortnameLinksModel
-from pay_api.models import EFTShortnames as EFTShortNamesModel
+from pay_api.models import EFTCreditInvoiceLink as EFTCreditInvoiceLinkModel
 from pay_api.models import Invoice as InvoiceModel
 from pay_api.models import InvoiceReference as InvoiceReferenceModel
-from pay_api.models import Payment as PaymentModel
 from pay_api.models import PaymentAccount as PaymentAccountModel
-from pay_api.models import PaymentLineItem as PaymentLineItemModel
 from pay_api.models import Receipt as ReceiptModel
 from pay_api.models import db
-from pay_api.services import EFTShortNamesService
 from pay_api.services.cfs_service import CFSService
-from pay_api.services.receipt import Receipt
 from pay_api.utils.enums import (
-    CfsAccountStatus, EFTShortnameStatus, InvoiceReferenceStatus, InvoiceStatus, PaymentMethod, PaymentSystem,
+    CfsAccountStatus, EFTCreditInvoiceStatus, InvoiceReferenceStatus, InvoiceStatus, PaymentMethod, PaymentSystem,
     ReverseOperation)
 from sentry_sdk import capture_message
 
 
-@dataclass
-class EFTShortnameInfo:
-    """Consolidated EFT Short name information for processing."""
-
-    id: int
-    auth_account_id: str
-    version: int
-
-
 class ElectronicFundsTransferTask:  # pylint:disable=too-few-public-methods
     """Task to link electronic funds transfers."""
 
     @classmethod
-    def link_electronic_funds_transfers(cls):
-        """Create invoice in CFS.
+    def _get_eft_credit_invoice_links_by_status(cls, status: str) \
+            -> List[InvoiceModel, EFTCreditInvoiceLinkModel, CfsAccountModel]:
+        """Get electronic funds transfer by state."""
+        query = db.session.query(InvoiceModel, EFTCreditInvoiceLinkModel, CfsAccountModel) \
+            .join(EFTCreditModel, EFTCreditModel.id == EFTCreditInvoiceLinkModel.eft_credit_id) \
+            .join(PaymentAccountModel, PaymentAccountModel.id == EFTCreditModel.payment_account_id) \
+            .join(CfsAccountModel, CfsAccountModel.account_id == PaymentAccountModel.id) \
+            .join(InvoiceModel, InvoiceModel.id == EFTCreditInvoiceLinkModel.invoice_id) \
+            .filter(CfsAccountModel.status == CfsAccountStatus.ACTIVE.value) \
+            .filter(CfsAccountModel.payment_method == PaymentMethod.EFT.value) \
+            .filter(EFTCreditInvoiceLinkModel.status_code == status) \
+            .order_by(InvoiceModel.payment_account_id, EFTCreditInvoiceLinkModel.id)
+        if status == EFTCreditInvoiceStatus.PENDING.value:
+            query = query.filter(InvoiceModel.disbursement_status_code.is_(None))
```

Comment:
Chucked this in here incase we have an invoice with a bad `disbursement_status_code`